platform/measurement_utils.cpp

```diff
@@ -189,4 +192,97 @@ string FormatSpeed(double metersPerSecond)
   return res;
 }
 
+bool OSMDistanceToMeters(string const & osmRawValue, double & outMeters)
+{
+  char * stop;
+  char const * s = osmRawValue.c_str();
+  outMeters = strtod(s, &stop);
+
+  // Not a number, was not parsed at all.
+  if (s == stop)
+    return false;
+
+  if (!isfinite(outMeters))
+    return false;
+
+  switch (*stop)
+  {
+  // Default units - meters.
+  case 0: return true;
+
+  // Feet and probably inches.
+  case '\'':
```

Comment:
Юра, я не до конца понимаю, куда уже дальше вылизывать, а главное, какую проблему это решит. Тут есть баги или оверхед или какая-то другая проблема?

Текущая реализация, на мой взгляд, очень простая и надежная. Юнит тесты проходят. Код читаемый. Каждый кейз можно улучшать по-отдельности, при желании.

Но если у тебя есть готовое лучшее решение, напиши его, пожалуйста.

> On 18.3.2016, at 16.49, ygorshenin <notifications@github.com> wrote:
> 
> In platform/measurement_utils.cpp:
> 
> > +  outMeters = strtod(s, &stop);
> > +
> > +  // Not a number, was not parsed at all.
> > +  if (s == stop)
> > +    return false;
> > +
> > +  if (!isfinite(outMeters))
> > +    return false;
> > +
> > +  switch (*stop)
> > +  {
> > +  // Default units - meters.
> > +  case 0: return true;
> > +
> > +  // Feet and probably inches.
> > +  case '\'':
> Я предлагаю очень простой и читаемый код разнести по двум очень простым и читаемым функциям - их легче каждую по отдельности проверить и у тебя останется логика, которая их использует как инструменты, а не высокоуровневая логика в перемешку с парсом.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>